### PR TITLE
drivers: eth: mcux: Prevent PHY entering factory test mode (i.MX specific)

### PR DESCRIPTION
**Pull request by armandciejak**
Force PHY normal operation in `eth_mcux_phy_setup` in case strap-in pins configure the PHY in factory test mode.

Depends on zephyrproject-rtos/hal_nxp#34

### CODE REVIEW

**Review by agansari — APPROVED** [review]

**Comment by pfalcon** [comment]
As mentioned in https://github.com/zephyrproject-rtos/hal_nxp/pull/34#pullrequestreview-373042497 , git commits/sign-offs should have a real email address (the idea is that someone may have questions or need help regarding the changes, so their original author should be reachable).

**Comment by armandciejak** [comment]
@pfalcon I changed my email address in the signed-off-by.

File: west.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ manifest:
       revision: 4bff01084d225996e4aae84b98be5969e2f9f33d
       path: tools/net-tools
     - name: hal_nxp
-      revision: 3e3f20e7de7606d735cdeefe21d5f2a51bf06095
+      revision: pull/34/head
```
Comment (MaureenHelm) [inline west.yml]: Please update to the new SHA now that zephyrproject-rtos/hal_nxp#34 has been merged

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
> @pfalcon I changed my email address in the signed-off-by.

Thank you

File: west.yml (hunk 1 above)
Comment (armandciejak) [reply west.yml]: I've done it, rebased the branch on master and fixed the conflicts. It is ready to be merged.

**Comment by armandciejak** [comment]
I've fixed a typo and tested on MIMXRT1060-EVK. It should now be ready to be merged.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**